File: oop.java
class Animal{
    public Animal(){
        System.out.println("Create Animal");
    }
    void sound(){
        System.out.println("Animal makes a sound");
    }
}

class Dog extends Animal{
    public Dog(){
        System.out.println("Create Dog");
    }
    
    @Override
    void sound(){
        System.out.println("Dog barks");
    }
}

class Cat extends Animal{
    public Cat(){
        System.out.println("Create Cat");
    }
    
    @Override
    void sound(){
        System.out.println("Cat meows");
    }
}


public class oop{
    public static void main(String[] args) {
        Animal a = new Animal();
        a.sound();

        Dog d = new Dog();
        d.sound();

        Cat c = new Cat();
        c.sound();
    }
}

File: oop.cpp
#include <iostream>
using namespace std;

class Animal {
public:
    Animal() { cout << "Create Animal\n"; }
    virtual ~Animal() { cout << "Delete Animal\n"; }
    virtual void sound() { cout << "Animal makes a sound\n"; }
};

class Dog : public Animal {
public:
    Dog() { cout << "Create Dog\n"; }
    ~Dog() { cout << "Delete Dog\n"; }
    void sound() override { cout << "Dog barks\n"; }
};

class Cat : public Animal {
public:
    Cat() { cout << "Create Cat\n"; }
    ~Cat() { cout << "Delete Cat\n"; }
    void sound() override { cout << "Cat meows\n"; }
};

int main() {
    Animal a;
    a.sound();

    Dog d;
    d.sound();

    Cat c;
    c.sound();

    return 0;
}
